Add product description for new teams map (#38660)
DataDog/documentation: content/en/account_management/teams/manage.md

A team that is managed externally from an identity provider displays a **Managed** badge in its header.

> ## Team hierarchies

Reviewer: **Two blocking items on this heading change.**

**1. The GovCloud warning was dropped.** `master` has this immediately under the old heading:

```
{{< site-region region="gov,gov2" >}}
<div class="alert alert-warning">
This feature is not supported for your selected Datadog site ({{< region-param key="dd_site_name" >}}).
</div>
{{< /site-region >}}
```

It isn't re-added anywhere in this PR. If hierarchies still aren't supported on `gov`/`gov2`, removing this actively misleads US1-Fed customers. Could you confirm whether Teams Map shipped to gov sites, or whether this came out accidentally during the rewrite?

**2. Renaming the heading breaks four inbound links.** The anchor changes from `#subteams-hierarchical-teams` to `#team-hierarchies`, and these four pages point at the old one:

- `content/en/account_management/guide/teams-and-access.md:43`
- `content/en/internal_developer_portal/_index.md:97`
- `content/en/internal_developer_portal/eng_reports/_index.md:46`
- `content/en/internal_developer_portal/catalog/set_up/ownership.md:65`

They all need updating in this PR, otherwise those links land on the page but not the section.

> Build a Teams map to visualize team hierarchies that let you nest teams within each other (subteams) to mirror your company's actual structure, giving you a more complete and accurate ownership model instead of a flat, disconnected list of teams. A team's related teams, parent teams, sister teams, and subteams, appear in its Teams Page, in the Hierarchy section of its Info tab. To explore the full map of all teams in your organization, visit the [Teams Map][7] in the Datadog Internal Developer Portal (IDP).

Reviewer: A few things in this paragraph:

- **Stray comma splits the subject from its verb**: "related teams, parent teams, sister teams, and subteams**,** appear in..."
- **The first sentence runs about 40 words** and ends in a trailing participial phrase ("...giving you a more complete and accurate ownership model instead of..."). That pattern is one of the AI-writing markers the style guide calls out.
- `visit` should be `see` (Vale flags this).
- **`Hierarchy` lost its bold.** `master` had `**Hierarchy**`; this page bolds UI elements elsewhere.
- "Teams Page" reads oddly capitalized — likely "team page".
- `(IDP)` is expanded and then never used again on the page, so the parenthetical can go.

```suggestion
Build a Teams map to visualize team hierarchies, nesting teams within each other (subteams) to mirror your company's actual structure. The result is a more complete ownership model than a flat, disconnected list of teams. A team's related teams, parent teams, sister teams, and subteams appear on its team page, in the **Hierarchy** section of the **Info** tab. To explore the full map of all teams in your organization, see the [Teams Map][7] in the Datadog Internal Developer Portal.
```

One question on content: **"sister teams" is new terminology** that doesn't appear on `master` and isn't defined here. Does the UI use that label? If not, worth aligning with whatever it does say.

{{< img src="/account_management/teams/github/manage/teams-map.png" alt="Teams map showing a hierarchy of parent and child teams, with a selected team's details panel displaying its members and subteams" style="width:100%;" >}}

With hierarchies defined, you can:

- **Escalate issues**: When an issue needs to go beyond the owning team, hierarchy makes it easy to route it up to the parent team for broader visibility or faster resolution, without having to manually track down who's next in the chain

> - **Roll-up filtering**: select a parent team to instantly see all data connected to any of its subteams, without manually selecting each one. Useful for director/VP-level views where you care about an entire org's footprint, not just one team's.

Reviewer: This list has four small consistency issues:

- **Capitalization after the colon** differs (`When` vs `select`). Definition-style items capitalize the first word of the description.
- **End punctuation** differs — no period on the first, period on the second. If any item is a full sentence, all take periods.
- `easy` and `just` are both on the removal list (Vale flags both).
- `director/VP-level` uses `/` to mean "and/or", which Datadog style avoids, and `org's` should be `organization's`.

```suggestion
- **Escalate issues**: When an issue needs to go beyond the owning team, hierarchy routes it up to the parent team for broader visibility and faster resolution, without manually tracking down who's next in the chain.
- **Roll-up filtering**: Select a parent team to see all data connected to any of its subteams, without selecting each one. Useful for director and VP-level views, where you care about an entire organization's footprint rather than one team's.
```

[3]: /account_management/scim/
[5]: /api/latest/teams/#add-a-member-team
[6]: /incident_response/on-call/

> [7]: https://app.datadoghq.com/software?env=%2A&fromUser=true&refresh_mode=sliding&selectedComponent=team&start=1785444148521&end=1785447748521

Reviewer: **This URL has a hardcoded one-hour time window and needs replacing.**

The `start` and `end` values decode to **2026-07-30 20:42:28 → 21:42:28 UTC**, so every reader who clicks it is pinned to that single hour, permanently. `fromUser=true`, `refresh_mode=sliding`, and `selectedComponent=team` are also copy-paste artifacts from a browser session.

A clean app link without the query string is what's needed here. The same URL appears as `[40]` in #38661, so both need the fix.

Separately: **`[5]` is now orphaned.** Removing the "use the [Teams APIs][5]" sentence leaves `[5]: /api/latest/teams/#add-a-member-team` defined but unused (it's used once on `master`, zero times here). Either drop the definition or bring the API pointer back — worth reading alongside #38661, which adds a Teams API link on the other page.
